### PR TITLE
Add filtering to institution location search

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
Related https://github.com/department-of-veterans-affairs/vets-website/pull/17455
## Original issue(s)
department-of-veterans-affairs/va.gov-team#23597

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Comment by voidspooks** [comment]
Please fix tests prior to asking for a review. Thank you.

**Review by delli-gatti-michael — APPROVED** [review]

**Comment by zurbergram** [comment]
@cbtkc This is ready for review

File: app/models/weam.rb [hunk 1]
```diff
@@ -82,8 +90,8 @@ class Weam < ImportableRecord
 
   def institution_type
     msg = 'Unable to determine institution type'
-    errors.add(:institution_type, msg) unless ['OJT', 'PRIVATE', 'FOREIGN', 'CORRESPONDENCE',
```
Comment (LindseySaari) [inline app/models/weam.rb:85]: Are the quotes supposed to be here? Seems like you want the constant (which evaluates to what you have in quotes)

Comment (zurbergram) [reply app/models/weam.rb:85]: I switched it to an array of constants representing each of the previous values.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**